### PR TITLE
Update vim highlights

### PR DESCRIPTION
**Pull request by vigoux**
Update vim highlights to include @Fymyte changes.
Also update the vim parser to ABI 14.

### CODE REVIEW

**Comment by vigoux** [comment]
Hmm the CI failure seems to be because of the nightly version of nvim not already being upated

**Review by clason — APPROVED** [review]

File: lockfile.json [hunk 1]
```diff
@@ -267,7 +267,7 @@
     "revision": "8f6b1f357d1231c420404b5f7a368a73c25adfa2"
   },
   "vim": {
-    "revision": "8296f301b19fc5f05820088b5b8b77e3edb5f526"
+    "revision": "6c7e93bdff915a423f546e0be438ecdd187d8a59"
```
[the remark this answers is not in the document]
Comment (vigoux) [reply lockfile.json:270]: Then we need to add that to the CI.
The question is how to do so.

[the remark this answers is not in the document]
Comment (clason) [reply lockfile.json:270]: You could just add an `install_from_grammar` field for now to unblock this PR? (Once https://github.com/nvim-treesitter/nvim-treesitter/pull/2328 is merged.)

[the remark this answers is not in the document]
Comment (clason) [reply lockfile.json:270]: Which it is -- try rebasing.

[the remark this answers is not in the document]
Comment (clason) [reply lockfile.json:270]: The alternative is to regenerate the upstream parser with ABI 13 and wait before bumping it to 14 again?

**Review by clason — APPROVED** [review]

**Pull request merged**